lib/src/main/java/io/ably/lib/http/AsyncHttp.java

```diff
@@ -147,6 +148,8 @@ private AblyRequestWithFallback(
 		public void run() {
 			int retryCountRemaining = Hosts.isRestFallbackSupported(http.host) ? http.options.httpMaxRetryCount : 0;
 			String candidateHost = http.host;
+			if(http.options.fallbackHosts != null)
+				Collections.shuffle(http.options.fallbackHosts);
```

Comment:
Piotr said:
I reverted my changes with ClientOptions#fallbackHosts back to the public field and shuffling it before every usage, that way we can be sure that every time someone use this field, it'll be random. If we randomize it only on the init, it can be overwritten later in the code since it's a public field and host order won't be random anymore. It also ensures that default hosts will also be randomized.